### PR TITLE
ARROW-7292: [CI] [C++] Add ASAN / UBSAN run

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@kszucs

File: cpp/src/arrow/util/bit_util.h [hunk 1]
```diff
@@ -999,8 +999,10 @@ class ARROW_EXPORT Bitmap : public util::ToStringOstreamable<Bitmap>,
     }
 
     // load remaining bits
-    SafeLoadWords(bitmaps, 0, bit_length, false, &visited_words);
-    visitor(visited_words);
+    if (bit_length > 0) {
```
Comment (pitrou) [inline cpp/src/arrow/util/bit_util.h]: @bkietz Does this change look ok?

**Review by kszucs — APPROVED** [review]
LGTM

**Review by kou — APPROVED** [review]
+1

**Comment by wesm** [comment]
I'd like to merge this but I can't log into JIRA right now for some reason, I contacted INFRA about it

**Pull request closed without merging**